sdk/include/opentelemetry/sdk/common/http_client.h

```diff
@@ -0,0 +1,163 @@
+#pragma once
+
+#include <chrono>
+#include <string>
+#include <vector>
+#include "opentelemetry/nostd/function_ref.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/version.h"
+
+/*
+ Usage Example
+
+ struct SimpleReponseHandler: public ResponseHandler {
+      void OnResponse(Response& res) noexcept override
+      {
+           if (res.IsSuccess()) {
+            res.GetNextHeader([](nostd::string_view name, std::string value) -> bool {
+                std::cout << "Header Name:" << name << " Header Value:"<< value ;
+                return true;
+            });
+            .. process response body
+           }
+      }
+
+      void OnError(nostd::string_view err) noexcept override
+      {
+          std::cout << " Error:" << err;
+      }
+  };
+
+  SessionManager sessionManager; // implementer can provide singleton implementation for it
+  auto session = sessionManager.createSession("localhost", 8000);
+  auto request = session->CreateRequest();
+  request->AddHeader(..);
+  SimpleResponseHandler res_handler;
+  session->SendRequest(res_handler);
+  session->FinishSession() // optionally in the end
+  ...shutdown
+  sessionManager.FinishAllSessions()
+*/
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace common
+{
+namespace http
+{
+
+enum class Method
+{
+  Get,
+  Post,
+  Put,
+  Options,
+  Head,
+  Patch,
+  Delete
+};
+
+enum class SessionState
+{
+  Created,             // session object is created
+  Ongoing,             // session is ongoing
+  Finished,            // session is finished ( this needs to be the final state )
+  Queued,              // http request is queued
+  TimedOut,            // Request timedout, no response received
+  Aborted,             // http request aborted due to local error,
+  Cancelled,           // http request cancelled, possibly due to session->CancelSession();
+  SendingFailed,       // http request sending failed
+  NetworkError,        // network error
+  SSLHandshakeFailed,  // ssl handshake failed
+  ReadError,           // error while reading response
+  WriteError           // error while writing rquest
+};
+
+using Byte           = uint8_t;
+using StatusCode     = uint16_t;
+using Body           = std::vector<Byte>;
+using SSLCertificate = std::vector<Byte>;
+
+class Request
+{
+public:
+  virtual void SetMethod(Method method) noexcept = 0;
+
+  virtual void SetUri(const nostd::string_view &uri) noexcept = 0;
```

Comment:
Same here: we can omit the const reference:

```suggestion
  virtual void SetUri(nostd::string_view uri) noexcept = 0;
```